src/ckb-daemon/input_mac.c

```diff
@@ -25,15 +30,29 @@ static void postevent(io_connect_t event, UInt32 type, NXEventData* ev, IOOption
     }
 
     IOGPoint location = {0, 0};
-    if((options & kIOHIDSetRelativeCursorPosition) && type != NX_MOUSEMOVED){
-        // Hack #2: IOHIDPostEvent will not accept relative mouse coordinates for any event other than NX_MOUSEMOVED
-        // So we need to get the current absolute coordinates from CoreGraphics and then modify those...
-        CGEventRef cge = CGEventCreate(nil);
-        CGPoint loc = CGEventGetLocation(cge);
-        CFRelease(cge);
-        location.x = floor(loc.x + ev->mouseMove.dx);
-        location.y = floor(loc.y + ev->mouseMove.dy);
-        options = (options & ~kIOHIDSetRelativeCursorPosition) | kIOHIDSetCursorPosition;
+
+    if(!checked_for_osx_sierra){
+        char osx_version_buf[256];
+        size_t size = sizeof(osx_version_buf);
+        sysctlbyname("kern.osrelease", osx_version_buf, &size, NULL, 0);
+        char osx_version[2];
+        strncpy(osx_version, osx_version_buf, 2);
+        osx_sierra = atoi(osx_version) >= 16;
```

Comment:
This is wrong. For example, `10.12.1` is built on XNU `15.6.0`. Moreover, trying to detect OS based on the kernel version is wrong - they don't depend on each other. I think we will have to find a better way. I tested different constants and variables exposed by `CTL_KERN`, to no avail. And that's reasonable, since kernel can't know about macOS version (which is done in a BSD style, not in a Linux style).

Reply:
I am referring to https://github.com/dylanaraps/neofetch/issues/433. Even though [Wikipedia](https://en.wikipedia.org/wiki/Darwin_%28operating_system%29#Release_history) says it should be 16.0, so Wikipedia must be wrong, too.

Reply:
Very good point. I'll do some more digging over the weekend and try to find a better way.

Reply:
I do know that we can shell out to `sw_vers -productVersion` to get the version number, but that's hacky and there's got to be a better way.

Reply:
TLDR; this is correct, sorry.

So, I took another approach. I went to opensource.apple.com, chose macOS 10.12 and looked at the XNU package version, which is `3789.1.32`. Great, thought I, this should be _the_ way to check for the kernel, since I would know exactly which kernel build is supplied with 10.12.0. I downloaded it to determine how this version is even generated and what does it have to do with the normal version. The answer is in `libkern/libkern/version.h.template`:

```c
/* version is a string of the following form, as returned by uname -v:
 * "Darwin Kernel Version <osrelease>: <build date>; <osbuilder>:<build root>"
 */
extern const char version[];
```

A `ripgrep "version\["` revealed the existence of `config/version.c`, which contains a comment:
```c
/* version.c
 * This file is a C source template for version.c, which is generated
 * on every make of xnu.  This template is processed by the script
 * xnu/config/newvers.pl based on the version information in the file
 * xnu/config/MasterVersion.
 */
```

which in turn led to `config/MasterVersion`:
```pl
16.0.0

# The first line of this file contains the master version number for the kernel.
# All other instances of the kernel version in xnu are derived from this file.
#
# The format of the version number must conform to the version resource format
# as described in TN1132: http://developer.apple.com/technotes/tn/tn1132.html
#
# In particular, the string is formatted as: J[.N[.R[S[L]]]], where:
#  J represents the kernel major version number (integer)
#  N represents the kernel minor version number (integer)
#  R represents the kernel revision number (integer)
#  S represents the kernel build stage (one of "d", "a", "b", or "r")
#  L represents the kernel pre-release level (integer)
#
# The correct way to make use of the kernel version within kernel code or a
# kext is to include libkern/verison.h.  version.h contains defines that can
# be used for build-time version logic and prototypes for variables that can
# be used for run-time version logic.
```

So, this confirms that `16` is the correct major version. I also downloaded `xnu-3248.60.10` supplied with `10.11.6` and `config/MasterVersion` contained `15.6.0`.

Funnily enough, if you try to actually follow the technical note link, you'll find nothing. Fortunately, there's a [mirror](http://mirror.informatimago.com/next/developer.apple.com/technotes/tn/tn1132.html) of it. Apple's versioning system is an entirely different animal.